framework/src/consumption/lib/redshift-serverless/redshift-serverless-namespace.ts

```diff
@@ -0,0 +1,185 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+import { randomBytes } from 'crypto';
+import { Names, RemovalPolicy, Stack, Tags } from 'aws-cdk-lib';
+import { Effect, IRole, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { Key } from 'aws-cdk-lib/aws-kms';
+import { ISecret, Secret } from 'aws-cdk-lib/aws-secretsmanager';
+import { AwsCustomResource, AwsCustomResourcePolicy, PhysicalResourceId } from 'aws-cdk-lib/custom-resources';
+import { Construct } from 'constructs';
+import { RedshiftServerlessNamespaceProps } from './redshift-serverless-namespace-props';
+import { Context, TrackedConstruct, TrackedConstructProps } from '../../../utils';
+
+/**
+ * Create a Redshift Serverless Namespace with the admin credentials stored in Secrets Manager
+ * @see https://awslabs.github.io/data-solutions-framework-on-aws/docs/constructs/library/Consumption/redshift-serverless
+ *
+ * @example
+ * const namespace = new dsf.consumption.RedshiftServerlessNamespace(this, 'DefaultServerlessNamespace', {
+ *    dbName: 'defaultdb',
+ *    name: 'default'
+ * });
+ */
+export class RedshiftServerlessNamespace extends TrackedConstruct {
+  /**
+   * Created namespace
+   */
+  readonly cfnResource: AwsCustomResource;
+
+  /**
+   * The name of the created namespace
+   */
+  readonly namespaceName: string;
+
+  /**
+   * The created Secrets Manager secret containing the admin credentials
+   */
+  readonly adminSecret: ISecret;
+
+  /**
+   * The roles attached to the namespace
```

Comment:
Can we describe a bit more what these roles are doing?
Maybe we should provide the keys of each role to be accessed?